data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -1808,6 +1966,29 @@ public Class<? extends Annotation> annotationType() {
         return SqlQueryConfiguration.DialectConfiguration.class;
     }
 
+    /**
+     * Hash a constraint name using MD5. Convert the MD5 digest to base 35
+     * (full alphanumeric), guaranteeing
+     * that the length of the name will always be smaller than the 30
+     * character identifier restriction enforced by a few dialects.
+     *
+     * @param s The name to be hashed.
+     *
+     * @return String The hashed name.
+     */
+    private static String hashedName(String s) {
```

Comment:
Saw this used by Hibernate and it's pretty useful since always creates the same FK generated id and can be used to drop and create reliably, plus it has length limit so wouldn't cause issue in the db.